tensorflow_addons/layers/embeddingbag.py

```diff
@@ -0,0 +1,130 @@
+
+def embeddingbag(
+    indices,
+    values,
+    weights,
+    name=None,
+):
+    """EmbeddingBag computation
+
+    See [PyTorch op](https://pytorch.org/docs/stable/generated/torch.nn.EmbeddingBag.html).
+
+    Equivalent to tf.gather() followed by tf.reduce_sum() across the last dimension, with optional
+    weights. Fusing these into a single op has massive benefits for execution speed and particularly
+    memory usage, as the intermediate output of the gather never needs to be materialized.
+
+    Args:
+      indices: An int32 or int64 `Tensor` of the indices to gather from
+          `values`. Must be at least 2-dimensional, as the last dimension
+          will be summed out. Maximum value must be less than values.shape[0].
+      values: A float32 `Tensor` from which to gather values. Must be rank 2.
+      weights: A float32 `Tensor` of weights which will be applied to each of
+          the gathered embedding vectors before the sum step.
+      name: A name for the operation (optional).
+
+    Returns:
+      A `Tensor` of the format specified by `data_format`.
+    """
+
+    # TODO Assert weights have same shape as indices, if present
+    with tf.name_scope(name or "embedding_bag"):
+        op_call = _embeddingbag_so.ops.addons_embedding_bag
+
+        if weights is None:
+            weights = tf.ones_like(indices, dtype=tf.float32)
+
+        ret = op_call(
+            indices,
+            values,
+            weights
+        )
+        return ret
+
+
+@tf.RegisterGradient("Addons>EmbeddingBag")
+def _embedding_bag_grad(op, grad):
+    indices, values, weights = op.inputs[:3]
+    op_call = _embeddingbag_so.ops.addons_embedding_bag_grad
+    values_grad, weights_grad = op_call(indices, values, weights, grad)[:2]  # Drop the dummy outputs
+    return [None, values_grad, weights_grad]
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class EmbeddingBag(tf.keras.layers.Layer):
+    """EmbeddingBag Layer
+
+    See [PyTorch op](https://pytorch.org/docs/stable/generated/torch.nn.EmbeddingBag.html).
+
+    Equivalent to tf.gather() followed by tf.reduce_sum() across the last dimension, with optional
+    weights. Fusing these into a single op has massive benefits for execution speed and particularly
+    memory usage, as the intermediate output of the gather never needs to be materialized.
+
+    Input Shapes:
+      indices: An int32 or int64 `Tensor` of the indices to gather from
+          `values`. Must be at least 2-dimensional, as the last dimension
+          will be summed out. Maximum value must be less than values.shape[0].
+      values: A float32 `Tensor` from which to gather values. Must be rank 2.
+      weights: A float32 `Tensor` of weights which will be applied to each of
+          the gathered embedding vectors before the sum step.
+
+    Output shape:
+        indices.shape[:-1], values.shape[-1]
+    """
+
+    def __init__(self, num_embeddings, embedding_dim, embeddings_initializer=None,
```

Comment:
in keras semantics:

- num_embeddings -> input_dim
- embedding_dim -> output_dim

https://www.tensorflow.org/api_docs/python/tf/keras/layers/Embedding

Reply:
Will fix!

Reply:
Fixed.